File: project-in-java/backend/Vehicle_Configurator-main/src/main/java/com/example/entity/Alternate_Component_Master.java
package com.example.entity;



import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import lombok.Data;

@Entity
@Data
public class Alternate_Component_Master {
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private int alt_id;
	private double delta_price;
	@ManyToOne
	@JoinColumn(name = "model_id")
	@JsonIgnore
	private Model model;
	
	@ManyToOne
	@JoinColumn(name = "comp_id")
	@JsonIgnore
	private Component component;
	
	@ManyToOne
	@JoinColumn(name="alt_comp_id")
	private Component altComponent;
}

File: project-in-dotnet/Vehicle_Configurator_Backend/Vehicle_Configurator/Domain/Entities/AlternateComponentMaster.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Vehicle_Configurator.Domain.Entities
{
    [Table("alternate_component_master")]
    public class AlternateComponentMaster
    {
        [Key]
        [Column("alt_id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int AltId { get; set; }

        [Column("delta_price")]
        public double DeltaPrice { get; set; }

        [Column("model_id")]
        public int ModelId { get; set; }
        public Model Model { get; set; }

        [Column("comp_id")]
        public int CompId { get; set; }
        public Component Component { get; set; }

        [Column("alt_comp_id")]
        public int AltCompId { get; set; }
        public Component AltComponent { get; set; }
    }
}
